File: demos/ios/Demos/Demos/views/BasePieChart.m


#import "BasePieChart.h"

@implementation BasePieChart

- (void)chartRender {    
    NSString *jsonPath = [[NSBundle mainBundle] pathForResource:@"Res/mockData_basePie" ofType:@"json"];
    NSString *jsonData = [NSString stringWithContentsOfFile:jsonPath encoding:NSUTF8StringEncoding error:nil];
    self.chart.canvas(self.canvasView).padding(0, 0, 0, 0.f);
    self.chart.source([F2Utils toJsonArray:jsonData]);
    self.chart.axis(@"percent", @{@"line": @(NO), @"label": @(NO), @"grid": @(NO)});
    self.chart.axis(@"a", @{@"line": @(NO), @"label": @(NO) , @"grid": @(NO)});
    self.chart.coord(@{@"type": @"polar", @"transposed": @(YES)});
    self.chart.interval().position(@"a*percent").color(@"name", @[]).adjust(@"stack").fixedSize(1).style(@{@"lineWidth": @(1),@"stroke": @"#ffffff"});
    self.chart.animate(@YES);
    self.chart.render();
}
@end

File: demos/ios/Demos-Swift/Demos-Swift/Views/BasePieChart.swift
//
//  BasePieChart.swift
//  F2Native
//
//  Created by weiqing.twq on 2024-06-09.
//

import F2

class BasePieChart: BaseLineChart {
    
    override func chartRender() {
        let jsonPath = Bundle.main.path(forResource: "Res/mockData_basePie", ofType: "json")
        guard let jsonString = try? String.init(contentsOfFile: jsonPath!) else {
            return
        }
        guard let chart = self.chart else {
            return
        }
        
        guard let canvasView = self.canvasView else {
            return
        }
        
        let jsonData = F2Utils.toJsonArray(jsonString)
        chart.clear()
        chart.canvas()(canvasView)
        chart.source()(jsonData)
        chart.padding()(10, 20, 10, 0)
        chart.axis()("percent", ["line": false, "label": false, "grid": false])
        chart.axis()("a", ["line": false, "label": false, "grid": false])
        chart.coord()(["type": "polar", "transposed": true])
        chart.interval()().position()("a*percent").color()("name", []).adjust()("stack").fixedSize()(1).style()(["lineWidth": 1, "stroke": "#ffffff"])
        chart.animate()(true)
        chart.render()()
    }
}

